packages/wallet/src/wallet.test.ts

```diff
@@ -61,6 +62,75 @@ describe('Pactus Wallet Tests', () => {
             expect(restoredWallet.getMnemonicWordCount()).toBe(12);
         });
 
+        it('should restore a wallet from external BIP39 mnemonic correctly', async () => {
+            // Generate a random BIP39 mnemonic with 12 words (128 bits of entropy)
+            const externalBip39Mnemonic = bip39.generateMnemonic(128);
+            // Verify the generated mnemonic is valid according to BIP39
+            expect(bip39.validateMnemonic(externalBip39Mnemonic)).toBe(true);
+            // Restore wallet using the external BIP39 mnemonic
+            const restoredWallet = Wallet.restore(core, externalBip39Mnemonic, 'test-password');
+            // Verify the restored wallet has the correct mnemonic
+            expect(restoredWallet.getMnemonic()).toBe(externalBip39Mnemonic);
+            expect(restoredWallet.getMnemonicWordCount()).toBe(12);
+
+            // Create an address to verify the wallet is functional
+            const address = restoredWallet.createAddress('Test Address');
+            expect(address).toBeTruthy();
+            expect(address.startsWith('pc1')).toBe(true);
+
+            // Verify the wallet can be used normally
+            const addresses = restoredWallet.getAddresses();
+            expect(addresses.length).toBe(1);
+            expect(addresses[0].publicKey).toBeTruthy();
+        });
+
+        it('should restore a wallet from external BIP39 mnemonic with 24 words correctly', async () => {
+            // Generate a random BIP39 mnemonic with 24 words (256 bits of entropy)
+            const externalBip39Mnemonic24 = bip39.generateMnemonic(256);
+            // Verify the generated mnemonic is valid according to BIP39
+            expect(bip39.validateMnemonic(externalBip39Mnemonic24)).toBe(true);
+
+            // Restore wallet using the external BIP39 mnemonic
+            const restoredWallet = Wallet.restore(core, externalBip39Mnemonic24, 'test-password');
```

Comment:
use this mnemonic as test:
`"abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon cactus"`

It should generate following addresses:

```
pc1rcx9x55nfme5juwdgxd2ksjdcmhvmvkrygmxpa3
pc1r7aynw9urvh66ktr3fte2gskjjnxzruflkgde94
```